auth/client.go

```diff
@@ -0,0 +1,134 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/futurehomeno/cliffhanger/hub"
+)
+
+// ProxyClientConfig is a configuration object for an authentication proxy client.
+type ProxyClientConfig struct {
+	PartnerCode string
+	Token       string
+	URL         string
+	Retry       int
+	RetryDelay  time.Duration
+}
+
+// setDefaults sets default configuration for a authentication proxy client.
+func (cfg *ProxyClientConfig) setDefaults() {
+	if cfg.Retry == 0 {
+		cfg.Retry = 5
+	}
+
+	if cfg.RetryDelay == 0 {
+		cfg.RetryDelay = 60 * time.Second
+	}
+
+	if cfg.URL == "" {
+		cfg.URL = ProxyURL(hub.EnvProd)
+	}
+}
+
+// ProxyClient is an interface representing a service responsible for utilization of the Partners API.
+type ProxyClient interface {
+	// ExchangeAuthorizationCode exchanges a one-time authorization code for the access token response.
+	ExchangeAuthorizationCode(code string) (*OAuth2TokenResponse, error)
+	// ExchangeRefreshToken exchanges a refresh token for the access token response.
+	ExchangeRefreshToken(refreshToken string) (*OAuth2TokenResponse, error)
+}
+
+// NewProxyClient creates new instance of a proxy proxyClient.
+func NewProxyClient(cfg *ProxyClientConfig) ProxyClient {
+	cfg.setDefaults()
+
+	return &proxyClient{
+		cfg:    cfg,
+		client: &http.Client{},
```

Comment:
I guess you should set some timeout here.

Reply:
Done. In future HTTP client should be injectable...